packages/scaffolding-cli/templates/build/azDevops/azure/templates/steps/build-csr.yml

```diff
@@ -0,0 +1,128 @@
+          customRegistry: ${{ parameters.customRegistry }}
+
+  # Install dependencies
+  - template: azDevOps/azure/templates/v2/steps/build-install-dependencies-node.yml@templates
+    parameters:
+      versionSpec: ${{ parameters.versionSpec }}
+      workingDirectory: ${{ parameters.project_root_dir }}
+      customRegistry: ${{ parameters.customRegistry }}
+      run_build: false
+
+  # Linting and formatting validation
+  - ${{ if eq(parameters.lint_formatting, true) }}:
+      - template: azDevOps/azure/templates/v2/steps/test-static-validate-node.yml@templates
+        parameters:
+          versionSpec: ${{ parameters.versionSpec }}
+          workingDirectory: ${{ parameters.project_root_dir }}
+          lintingSharedConfigPackage: ${{ parameters.shared_eslint_config }}
+          customRegistry: ${{ parameters.customRegistry }}
+
+  # Linting and unit tests
+  - ${{ if eq(parameters.unit_test, true) }}:
+      - template: azDevOps/azure/templates/v2/steps/test-unit-node.yml@templates
+        parameters:
+          versionSpec: ${{ parameters.versionSpec }}
+          workingDirectory: ${{ parameters.project_root_dir }}
+
+  # SonarCloud
+  - ${{ if eq(parameters.static_code_analysis, true) }}:
+      - template: azDevOps/azure/templates/v2/steps/test-static-code-sonar.yml@templates
+        parameters:
+          workingDirectory: ${{ parameters.project_root_dir }}
+
+  # Consumer driven contract tests
+  - ${{ if eq(parameters.contract_test, true) }}:
+      - template: azDevOps/azure/templates/v2/steps/test-contract-pact.yml@templates
+        parameters:
+          versionSpec: ${{ parameters.versionSpec }}
+          build_number: ${{ parameters.docker_imagetag }}
+          workingDirectory: ${{ parameters.project_root_dir }}
+
+  # Check with Pact broker if safe to proceed with deployment (breaking contract changes)
+  - ${{ if and(eq(parameters.contract_test, true), eq(parameters.contract_check_can_deploy, true)) }}:
+      - template: azDevOps/azure/templates/v2/steps/deploy-contract-pact-can-deploy.yml@templates
+        parameters:
+          workingDirectory: ${{ parameters.project_root_dir }}
+
+  # Functional tests running on locally built and hosted (undeployed)
+  - ${{ if eq(parameters.cypress_e2e_test, true) }}:
+      - template: azDevOps/azure/templates/v2/steps/test-functional-cypress.yml@templates
+        parameters:
+          env_vars: ${{ parameters.cypress_e2e_env_vars}}
+          workingDirectory: ${{ parameters.project_root_dir }}
+
+  # Visual regression tests running on locally built and hosted (undeployed)
+  - ${{ if eq(parameters.cypress_applitools_test, true) }}:
+      - task: ApplitoolsBuildTask@0
+        displayName: "Tests: Visual Regression with Applitools"
+
+  # Publish Artefacts if required
+  - ${{ if eq(parameters.functional_test, true) }}:
+      - task: PublishPipelineArtifact@1
+        displayName: "Publish: Functional Tests Artifact"
```

Comment:
We should prob create a ticket to get this naming consistant.

Reply:
👍